src/diffusers/models/controlnet_flux.py

```diff
@@ -270,9 +276,22 @@ def forward(
             if guidance is None
             else self.time_text_embed(timestep, guidance, pooled_projections)
         )
+
         encoder_hidden_states = self.context_embedder(encoder_hidden_states)
 
-        txt_ids = txt_ids.expand(img_ids.size(0), -1, -1)
+        if self.union:
+            # union mode
+            if controlnet_mode is None:
+                raise ValueError("`controlnet_mode` cannot be`None` when using union-controlnet")
+
+            # union mode emb
+            controlnet_mode_emb = self.controlnet_mode_embedder(controlnet_mode)
+            encoder_hidden_states = torch.cat([controlnet_mode_emb, encoder_hidden_states], dim=1)
+            txt_ids = torch.cat([txt_ids[:, :1], txt_ids], dim=1)
```

Comment:
we merged this https://github.com/huggingface/diffusers/pull/9074 that removed the batch dimension in txt_ids and img_ids

so after resolve the conflicts, you will also need update the code here I think (need to cat on dim 0 now, instead of 1)